system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -65,17 +106,43 @@ class TopSite extends React.Component {
             <span dir="auto">{title}</span>
           </div>
         </a>
-        <button className="context-menu-button" onClick={this.onMenuButtonClick}>
-          <span className="sr-only">{`Open context menu for ${title}`}</span>
-        </button>
-        <LinkMenu
-          dispatch={dispatch}
-          index={index}
-          onUpdate={this.onMenuUpdate}
-          options={TOP_SITES_CONTEXT_MENU_OPTIONS}
-          site={link}
-          source={TOP_SITES_SOURCE}
-          visible={isContextMenuOpen} />
+        {!editMode &&
+          <div>
+            <button className="context-menu-button" onClick={this.onMenuButtonClick}>
+              <span className="sr-only">{`Open context menu for ${title}`}</span>
+            </button>
+            <LinkMenu
+              dispatch={dispatch}
+              index={index}
+              onUpdate={this.onMenuUpdate}
+              options={TOP_SITES_CONTEXT_MENU_OPTIONS}
+              site={link}
+              source={TOP_SITES_SOURCE}
+              visible={isContextMenuOpen} />
+          </div>
+        }
+        {editMode &&
+          <div className="edit-menu">
+            {link.isPinned &&
```

Comment:
oops, yeah I guess everything else is the same. Oh, and the intl message id but I can do the same thing.